src/main/java/eu/xenit/gradle/docker/DockerBuildBehavior.java

```diff
@@ -154,21 +155,32 @@ private DockerBuildImage createDockerBuildImageTask(Project project,
         boolean automaticTags = dockerBuildExtension.get().getAutomaticTags();
 
         String jenkinsBranch = cleanForDockerTag(JenkinsUtil.getBranch());
+        if (jenkinsBranch.length() > 128) {
+            throw new Error(DOCKER_TAG_LENGTH_CONSTRAINT_ERRORMSG);
```

Comment:
Do not throw `Error`: https://docs.oracle.com/javase/8/docs/api/index.html?java/lang/Error.html I would propose creating a custom exception, or throwing `IllegalStateException` or a `GradleException`

These checks are also not quite in the right place: things are appended to the jenkins branch, so this length check is not necessary.